src/toxcore/dht/dht_node.rs

```diff
@@ -155,14 +155,36 @@ impl DhtNode {
                     }
                 },
                 None => {
-                    self.assoc4.saddr.map(Into::into)
+                    match self.assoc4.saddr {
```

Comment:
Did you rebase successfully?

Reply:
It is changed to leave warning because if it is None then something goes wrong.

Reply:
https://github.com/tox-rs/tox/blob/24cfbb1821fa9e71e4832544e3011f14c6fe8f73/src/toxcore/dht/dht_node.rs#L170-L173

Reply:
https://doc.rust-lang.org/std/option/enum.Option.html#method.or_else